chore(website): Add docs on OpenTelemetry and Jaeger (#12271)
cloudquery/cloudquery: website/pages/docs/advanced-topics/monitoring.md

---
title: Monitoring CloudQuery
description: Short walkthrough on how to use open telemetry to monitor CloudQuery Syncs
---

# Monitoring

Monitoring CloudQuery can be done in a number of main ways:

- Logging

> - Open-Telemetry

Reviewer: ```suggestion
- OpenTelemetry
```

## Logging

> CloudQuery is utilizing structured [logging](../reference/cli/cloudquery) (in plain and json formats) which can be analyzed by local tools such as `jq`, `grep` and remote aggregations tools like `loki`, `datadog` or any other popular log aggregation that supports structured logging.

Reviewer 2: ```suggestion
CloudQuery utilizes structured [logging](../reference/cli/cloudquery) (in plain and JSON formats) which can be analyzed by local tools such as `jq`, `grep` and remote aggregations tools like `loki`, `datadog` or any other popular log aggregation that supports structured logging.
```

## OpenTelemetry (Preview)

> ELT workloads can be long running and sometimes it is necessary to understand better what calls taking the most time to potentially optimize those on the plugin side, ignore them or split them to a different workload.

Reviewer: ```suggestion
ELT workloads can be long running and sometimes it is necessary to better understand what calls are taking the most time; to potentially optimize those on the plugin side, ignore them or split them to a different workload.
```

> CloudQuery supports [OpenTelemetry](https://opentelemetry.io/) tracing out of the box and can be enabled easily via [configuration](../reference/source-spec).

Reviewer 2: ```suggestion
CloudQuery supports [OpenTelemetry](https://opentelemetry.io/) tracing out of the box and can be enabled easily via [configuration](/docs//reference/source-spec).
```

We usually have absolute links so if we move this page to another location the link still works

To collect traces you need a collector that supports OpenTelemetry protocol, for example [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/). For example you can use [Jaeger](https://opentelemetry.io/docs/instrumentation/go/exporters/#jaeger) to visualize and analyze traces.

> To start jaeger locally you can use docker:

Reviewer: ```suggestion
To start Jaeger locally you can use Docker:
```

```bash
docker run -d \
  -e COLLECTOR_OTLP_ENABLED=true \
  -p 16686:16686 \
  -p 4318:4318 \
  jaegertracing/all-in-one:latest
```

and then specify in the source spec the following:

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]

>   enabled: true

Reviewer 2: ```suggestion
```

Is `enabled: true` part of the spec?
